base/multidimensional.jl

```diff
@@ -248,29 +262,36 @@ module IteratorsMD
 
     For cartesian to linear index conversion, see [`LinearIndices`](@ref).
     """
-    struct CartesianIndices{N,R<:NTuple{N,AbstractUnitRange{Int}}} <: AbstractArray{CartesianIndex{N},N}
+    struct CartesianIndices{N,R<:NTuple{N,OrdinalRange{Int, Int}}} <: AbstractArray{CartesianIndex{N},N}
         indices::R
     end
 
     CartesianIndices(::Tuple{}) = CartesianIndices{0,typeof(())}(())
```

Comment:
This seems a little bit inconsistant:

```julia
julia> CartesianIndices(())
0-dimensional CartesianIndices{0, Tuple{}}:
CartesianIndex()

julia> CartesianIndices(rand(0))
0-element CartesianIndices{1, Tuple{Base.OneTo{Int64}}}
```

Reply:
I think this inconsistency comes from the fact that:

```julia
julia> axes(())
(Base.OneTo(0),)

julia> length(())
0

julia> size(())
ERROR: MethodError: no method matching size(::Tuple{})
```

In any case, it should be fixed in other PRs.